### PR TITLE
[Python] Datetime and Categorical handling improvement

### PR DESCRIPTION
**Pull request by guillett**
Depends on #247.

* Allow null mask retrieval in Python
* Prevent int to float conversion when nulls are present (particularly relevant for high precision Datetimes - many significant figures).
* Prevent float array generation when nulls are present in categorical

### CODE REVIEW

File: python/feather/interop.h [hunk 1]
```diff
@@ -667,6 +668,55 @@ class FeatherDeserializer {
...
+  if (arr.null_count > 0) {
+    for (int64_t i = 0; i < arr.length; ++i) {
+      out_values[i] = util::get_bit(arr.nulls, i) ? 1 : 0;
+    }
+  } else {
+    for (int64_t i = 0; i < arr.length; ++i) {
+      out_values[i] = 1;
+    }
+  }
+  return out;
```
Comment (wesm) [inline python/feather/interop.h]: If you are only using this to set nulls, why not set 1 for null, 0 for not-null, and skip the invert steps (extra computation cost)?

**Comment by wesm** [comment]
Sorry, I merged #247, so this needs to be rebased. Thanks for digging through this

File: python/feather/interop.h (hunk 1 above)
Comment (guillett) [inline python/feather/interop.h]: You are right. I will change the function name to *null_mask*.

**Comment by wesm** [comment]
apparently i can rebase and push to your branch now

**Pull request closed without merging**